File: ginorst/src/insertion.rs
pub struct Sort<'a> {
    pub unsorted: &'a mut Vec<isize>,
}

impl Sort<'_> {
    pub fn new(arr: &mut Vec<isize>) -> Sort {
        Sort { unsorted: arr }
    }
    #[allow(dead_code)]
    pub fn sort(&mut self) {
        let arr = &mut self.unsorted;
        for i in 1..arr.len() {
            for j in 0..i {
                if arr[j] > arr[i] {
                    arr.swap(i, j);
                }
            }
        }
    }
    pub fn print(&self) {
        println!("{:?}", &self.unsorted);
    }
}

File: ts/insertion.ts
let sort = (arr: number[]) => {
    for (let i = 1; i < arr.length; i++) {
        for (let j = 0; j < i; j++) {
            if (arr[i] < arr[j]) {
                let temp = arr[i];
                arr[i] = arr[j];
                arr[j] = temp;
            }
        }
    }
    console.log(arr);
};

let arr: number[] = [
    8, 7, 4, -7, 1, 9, 3, -9, 5, -3, -6, 0, -5, -8, -4, 2, -10, -2, 6, -1, 10,
];

sort(arr);
